File: packages/ui-ios/Sources/ChorusUICatalog/ButtonCatalog.swift
import SwiftUI
import ChorusUI
import ChorusTokens

/// Catalog screen matching the docs site's Button stage. Each row is a
/// (flavor × appearance) cell exposed at all three sizes plus a disabled
/// + leadingIcon variant. Drop this into a SwiftUI preview or wire it
/// into your sample app's navigation to confirm visual parity with
/// `apps/docs/components/standard.md`.
public struct ChorusButtonCatalog: View {
    public init() {}

    public var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: ChorusLayout.Stack.lg) {
                section("Primary") {
                    appearanceRow(appearance: .primary, flavor: .standard)
                }
                section("Secondary") {
                    appearanceRow(appearance: .secondary, flavor: .standard)
                }
                section("Outlined") {
                    appearanceRow(appearance: .outlined, flavor: .standard)
                }
                section("Tertiary") {
                    appearanceRow(appearance: .tertiary, flavor: .standard)
                }
                section("Destructive · primary") {
                    appearanceRow(appearance: .primary, flavor: .destructive)
                }
                section("Destructive · outlined") {
                    appearanceRow(appearance: .outlined, flavor: .destructive)
                }
                section("Disabled") {
                    HStack(spacing: ChorusLayout.Inline.md) {
                        ChorusButton("Primary", appearance: .primary, action: {})
                            .disabled(true)
                        ChorusButton("Outlined", appearance: .outlined, action: {})
                            .disabled(true)
                    }
                }
                section("With leading icon") {
                    HStack(spacing: ChorusLayout.Inline.md) {
                        ChorusButton("Save", leadingIcon: Image(systemName: "checkmark"), action: {})
                        ChorusButton("Delete",
                                     appearance: .outlined,
                                     flavor: .destructive,
                                     leadingIcon: Image(systemName: "trash"),
                                     action: {})
                    }
                }
                section("Full width") {
                    ChorusButton("Continue", fullWidth: true, action: {})
                }
            }
            .padding(ChorusLayout.Page.md)
        }
    }

    @ViewBuilder
    private func section<Content: View>(_ title: String, @ViewBuilder content: () -> Content) -> some View {
        VStack(alignment: .leading, spacing: ChorusLayout.Stack.sm) {
            Text(title)
                .font(.system(size: ChorusTypography.Label.sm.size, weight: ChorusTypography.Label.sm.fontWeight))
                .foregroundStyle(.secondary)
            content()
        }
    }

    private func appearanceRow(
        appearance: ChorusButton.Appearance,
        flavor: ChorusButton.Flavor
    ) -> some View {
        HStack(alignment: .center, spacing: ChorusLayout.Inline.md) {
            ChorusButton("Large", size: .large, appearance: appearance, flavor: flavor, action: {})
            ChorusButton("Medium", size: .medium, appearance: appearance, flavor: flavor, action: {})
            ChorusButton("Small", size: .small, appearance: appearance, flavor: flavor, action: {})
        }
    }
}

#if DEBUG
struct ChorusButtonCatalog_Previews: PreviewProvider {
    static var previews: some View {
        Group {
            ChorusButtonCatalog().preferredColorScheme(.light)
            ChorusButtonCatalog().preferredColorScheme(.dark)
        }
    }
}
#endif

File: packages/ui-android/chorus-ui-catalog/src/main/kotlin/dev/blinddsai/chorus/ui/catalog/ButtonCatalog.kt
package dev.blinddsai.chorus.ui.catalog

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import dev.blinddsai.chorus.tokens.ChorusTheme
import dev.blinddsai.chorus.tokens.generated.ChorusLayout
import dev.blinddsai.chorus.ui.button.ChorusButton
import dev.blinddsai.chorus.ui.button.ChorusButtonAppearance
import dev.blinddsai.chorus.ui.button.ChorusButtonFlavor
import dev.blinddsai.chorus.ui.button.ChorusButtonSize

/**
 * Catalog screen mirroring the docs site Button stage. Each section is a
 * (flavor × appearance) cell shown at all three sizes; trailing sections
 * cover disabled, leading-icon, and fullWidth states. Drop into a sample
 * app or preview to validate visual parity.
 */
@Composable
fun ChorusButtonCatalog() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(ChorusLayout.Page.md),
        verticalArrangement = Arrangement.spacedBy(ChorusLayout.Stack.lg),
    ) {
        CatalogSection("Primary") { Row3(ChorusButtonAppearance.Primary, ChorusButtonFlavor.Standard) }
        CatalogSection("Secondary") { Row3(ChorusButtonAppearance.Secondary, ChorusButtonFlavor.Standard) }
        CatalogSection("Outlined") { Row3(ChorusButtonAppearance.Outlined, ChorusButtonFlavor.Standard) }
        CatalogSection("Tertiary") { Row3(ChorusButtonAppearance.Tertiary, ChorusButtonFlavor.Standard) }
        CatalogSection("Destructive · primary") { Row3(ChorusButtonAppearance.Primary, ChorusButtonFlavor.Destructive) }
        CatalogSection("Destructive · outlined") { Row3(ChorusButtonAppearance.Outlined, ChorusButtonFlavor.Destructive) }
        CatalogSection("Disabled") {
            Row(horizontalArrangement = Arrangement.spacedBy(ChorusLayout.Inline.md)) {
                ChorusButton("Primary", onClick = {}, enabled = false)
                ChorusButton("Outlined", onClick = {}, appearance = ChorusButtonAppearance.Outlined, enabled = false)
            }
        }
        CatalogSection("Full width") {
            ChorusButton("Continue", onClick = {}, fullWidth = true)
        }
    }
}

@Composable
private fun Row3(appearance: ChorusButtonAppearance, flavor: ChorusButtonFlavor) {
    Row(horizontalArrangement = Arrangement.spacedBy(ChorusLayout.Inline.md)) {
        ChorusButton("Large", onClick = {}, size = ChorusButtonSize.Large, appearance = appearance, flavor = flavor)
        ChorusButton("Medium", onClick = {}, size = ChorusButtonSize.Medium, appearance = appearance, flavor = flavor)
        ChorusButton("Small", onClick = {}, size = ChorusButtonSize.Small, appearance = appearance, flavor = flavor)
    }
}

@Preview(showBackground = true)
@Composable
private fun PreviewLight() {
    ChorusTheme(darkTheme = false) { ChorusButtonCatalog() }
}

@Preview(showBackground = true)
@Composable
private fun PreviewDark() {
    ChorusTheme(darkTheme = true) { ChorusButtonCatalog() }
}
